test/tools/cli/IonProcess.java

```diff
@@ -544,14 +502,28 @@ private static Event ionStreamToEvent(IonReader ionReader) {
                     event.setDepth(ionReader.intValue());
                     break;
             }
-        }
 
+        }
+        //compare text value and binary value
+        if (textValue != null && binaryValue != null) {
+            IonSystem ionSystem = IonSystemBuilder.standard().build();
+            IonValue text = ionSystem.singleValue(textValue);
+            IonValue binary = ionSystem.singleValue(binaryValue);
+            if (!Equivalence.ionEquals(text, binary)) {
+                throw new IonException("invalid Event: Text value and Binary value are different");
+            } else {
+                IonValue v = IonSystemBuilder.standard().build().singleValue(textValue);
+                event.setValue(v);
+            }
+        }
         ionReader.stepOut();
+
         return event;
     }
 
+
     private static ImportDescriptor[] ionStreamToImportDescriptors(IonReader ionReader) {
-        LinkedList<ImportDescriptor> imports = new LinkedList<>();
+        ArrayList<ImportDescriptor> imports = new ArrayList();
```

Comment:
When you're assigning to a list, just use `List` as the declared type unless there is something you absolutely need from the particular `List` implementation. That way if you change implementations you only have to make a change in one place. Also, always use `<>` to indicate that the type you're instantiating matches what was declared.

```java
List<ImportDescriptor> imports = new ArrayList<>();
```

Reply:
Changed.